### ISSUE TITLE
Bug in label normalization

### ISSUE THREAD

**Issue by abylouw** [CSTR-Edinburgh/merlin#434]
There is a bug in the label normalization for questions with only single context, for example:
``
CQS "Num-Phrases_in_Utterance"  	                {-(\d+)}
``
from questions-radio_dnn_416.hed

Take for example the following label from slt_arctic, arctic_a0060.lab
```
x^sil-eh+n=iy@1_2/A:0_0_0/B:1-1-2@1-3&1-3#1-2$1-2!0-2;0-2|eh/C:0+0+2/D:0_0/E:content+3@1+1&1+0#0+0/F:det_1/G:0_0/H:3=1@1=2|L-L%/I:6=6/J:9+7-2
```
This is the first non silence context label. The label vector value is calculated from: 
https://github.com/CSTR-Edinburgh/merlin/blob/6ab3ccc9e6a1f55cdec1c6ebcbf0f0f1c65e9422/src/frontend/label_normalisation.py#L791
which for the above label will return a ``1`` for this question, instead of ``2``. The regex matches on "/B:1-**1**-2/" instead of "/J:9+7-**2**". Therefore the number of phrases for each label line will not be consistent. There may be other examples of this.

I would suggest adding a start prefix and end prefix to the labels as well as the questions with no extra context, such as above and the "LL-" type questions.

**Comment by abylouw** [comment]
I see that the "LL-" type questions do get a prefix:
https://github.com/CSTR-Edinburgh/merlin/blob/6ab3ccc9e6a1f55cdec1c6ebcbf0f0f1c65e9422/src/frontend/label_normalisation.py#L870

I have created pull request #435 that fixes the last question issue (the one with only a left hand side context).

**Comment by ZackHodari** [comment]
As a note for anyone coming across this issue.

The reason this occurs is due to the definition of the question file. The question regexes should match with exactly one position in a label. If this is not the case then Merlin should not be relied on to fix it (though thanks to a PR[#435](https://github.com/CSTR-Edinburgh/merlin/pull/435) it does fix this for you).

- i.e. the following question that previously failed:
`CQS "Num-Phrases_in_Utterance" {-(\d+)}`

- should be defined as:
`CQS "Num-Phrases_in_Utterance" {*-(\d+)}`

Where the `*` indicates that there can be any number of characters before `-`, but no characters after `\d+`. Without any `*` characters in the question, `wildcards2regex` will not handle start and end of sequence behaviour in the python regex.
